### DISCUSSION TITLE
DISTRIBUTIONS.txt in Strawberry Perl  version 5.40.0.1

### DISCUSSION THREAD

**Discussion by sisyphus** [StrawberryPerl/Perl-Dist-Strawberry#217] [General]
Just some observations - which, I think, also apply to the listing in the release notes at https://github.com/StrawberryPerl/strawberryperl.com/commit/d09bb7c2500bebf8178259237a446745e5e5b5a9 .
If any of these observations call for the raising of  an "Issue", then just say so, and I'll oblige.
But there may well be good reasons that this entire post can simply be dismissed as lacking significance.

I've recently worked my way through building the modules listed in DISTRIBUTIONS.txt on a quadmath build of perl-5.40.0.
(That build used the gcc-13.2.0 toolset and libraries, as provided by Strawberry's 5.40.0.1-PDL release.)

There are 2 modules in vendor/lib (Devel-LexAlias-0.05 and Devel-Caller-2.07) that are not listed in DISTRIBUTIONS.txt.
Both of those modules seem to exist in "their own right" - that is, not as part of some general "Devel" package.
If there's a good reason to exclude them from the list, then I couldn't spot it ;-)

TimeDate is listed as "TimeDate-2.33", but it actually reports itself as being version 1.21.
(I think the 2,33 distro has been installed, but TimeDate.pm itself is stuck on version 1.21.)

In building these "vendor" modules on my quadmath build of perl, I just worked through the list in DISTRIBUTIONS.txt, running <code>cpan -i ...</code> where needed for each of the listings (modified appropriately to satisfy the cpan utility's requirements), but cpan complained that it "couldn't find a matching namespace", when it came to installing the following:
1) cpan-outdated-0.32 - I needed to run <code>cpan -i App::cpanoutdated</code>
2) libwww-perl-6.77 - I needed to run <code>cpan -i LWP</code>;
3) podlators-v6.0.2 - I needed to run <code>cpan -i Pod::Man</code>;
4) Scalar-List-Utils-1.65 - I needed to run <code>cpan -i List::Util</code>.

As you can see, those 4 listings don't readily convert to a format that the cpan utility can locate.

**Comment by shawnlaffan** [comment]
The set of modules installed as part of an SP build are in the config file.  For 5.40.0.1 these are at https://github.com/StrawberryPerl/Perl-Dist-Strawberry/blob/master/share/64bit-5.40.0.1.pp

The installation includes dependencies, but these seem not to be added to the list that is used to build DISTRIBUTIONS.txt.  
https://github.com/StrawberryPerl/Perl-Dist-Strawberry/blob/8ec82760ac883c7a6a00425fc2ec53967526b528/lib/Perl/Dist/Strawberry/Step.pm#L397-L429

Maybe they are in an exclude list but I have not checked, and I suspect not.
https://github.com/StrawberryPerl/Perl-Dist-Strawberry/blob/8ec82760ac883c7a6a00425fc2ec53967526b528/lib/Perl/Dist/Strawberry/Step/CreateReleaseNotes.pm#L31-L40


FWIW, Devel::LexAlias is a dependency of Devel::Caller.  I am not sure what in SP depends on Devel::Caller but it could be in this list or higher in the chain: https://metacpan.org/module/Devel::Caller/requires


In terms of the second part of the question:

LWP and CPAN::Outdated are explicitly installed.  

Pod::Man is a core module and is likely updated as part of the relevant step.  
https://github.com/StrawberryPerl/Perl-Dist-Strawberry/blob/8ec82760ac883c7a6a00425fc2ec53967526b528/share/64bit-5.40.0.1.pp#L124-L133

List::Util is part of the Scalar::List::Utils distribution on CPAN.  It is also a core module.  

FWIW, SP uses cpanm to install most packages.

**Comment by shawnlaffan** [comment]
I missed TimeDate.pm.  

The version mismatch is also on the TimeDate repo.  `$TimeDate::VERSION` has not been updated to match the rest of the distribution.  

e.g. compare https://github.com/atoomic/perl-TimeDate/blob/21b0250cdbf98ad01f6d5a447d99f10adaad9ac7/lib/TimeDate.pm#L5 and https://github.com/atoomic/perl-TimeDate/blob/21b0250cdbf98ad01f6d5a447d99f10adaad9ac7/lib/Time/Zone.pm#L51

**Comment by sisyphus** [comment]
Thanks Shawn.
The link to 64bit-5.40.0.1.pp enabled me to find the Crypt::SSLeay and PPM sources and I've now completed my "quadmath" exercise, including the building of PDL.

I've also since noticed that Inline::C and some of its prereqs are not listed in DISTRIBUTIONS.txt, despite being in vendor/lib.
However, I gather that you're not much bothered by such omissions - and I'm quite at ease with that.

**Reply by shawnlaffan** [reply]
No worries Rob.  

The DISTRIBUTIONS.txt is currently not updated for the PDL builds, which is where Inline::C and several other packages are installed.  That's being tracked under #203
